src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7788,6 +7794,7 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
             foreach (var scope in new ExtensionScopes(this))
             {
                 if (!left.Type.IsTypeParameter()
+                    && !left.Type.IsExtension
```

Comment:
>&& !left.Type.IsExtension [](http://example.com/codeflow?start=20&length=25)

Could you please elaborate? Why this change is needed? Why it is not needed in the next `if`?

Reply:
I'll move up and clarify. The logic for finding implicit extension type members asserts that we're not looking at an extension type. But we could hit it with `Extension.Missing`...
We disallow defining extensions and extension methods for extension types, so we can skip all this when we're looking at an extension type.